### DISCUSSION TITLE
Prisma Client Disconnect from Database upon Hot reload (Nuxt 3)

### DISCUSSION THREAD

**Discussion by chuahziyang** [prisma/prisma#14666] [Q&A]
### Bug description

I'm currently using nuxt 3 + Prisma in a project. Currently, whenever I hot reload my project, The Prisma client disconnects from the server (The connection dropped when checking on MYSQL workbench aka disappears from client connections list). Once the connection has dropped, nuxt3 refuses to process HTTP requests. The Prisma Client is configured as such:
```
import { PrismaClient } from "@prisma/client";

let prisma;

if (process.env.NODE_ENV === "production") {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }
  prisma = global.prisma;
}

export default prisma;
```

### How to reproduce

1. Clone https://github.com/HavenOfExcellence/Resource-Management-APp/
2. Set up local mysql server, and add the server connection string to .env
3. npm i
4. npm run dev, and navigate to localhost:3000/testing
5. save any server file/page file/component file to force a hot reload.
6. The connection has now dropped, refreshing localhost:3000 should result in a permanently loading page

### Expected behavior

The connection is expected not to drop on reload, the prisma client should continue to process requests even after a hot reload

### Prisma information

Schema
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model User {
  id       String  @id @default(uuid())
  name     String?
  email    String? @unique
  password String?
  Event    Event?  @relation(fields: [eventId], references: [id])
  eventId  String?
  hub      String?
}

model Event {
  id       String    @id @default(uuid())
  name     String?
  date     DateTime?
  startime String?
  endtime  String?
  fullday  Boolean
  users    User[]
}

```
.env 
```
DATABASE_URL=mysql://root:x2103811760@localhost:3306/ippt
```

The way Prisma is imported
```
import { PrismaClient } from "@prisma/client";

let prisma;

if (process.env.NODE_ENV === "production") {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }
  prisma = global.prisma;
}

export default prisma;

```

Sample Server Route
```
import prisma from "~~/utils/Prisma";
export default defineEventHandler(async (event) => {
  const user = await prisma.user.findMany({
    select: {
      name: true,
      id: true,
    },
  });
  console.log(user);
  console.log("asdasd");
  return user;
});

```

### Environment & setup

I have tested this 2 separate computers so it is unlikely an OS issue

### Prisma Version

```
prisma                  : 4.0.0
@prisma/client          : 4.0.0
Current platform        : windows
Query Engine (Node-API) : libquery-engine da41d2bb3406da22087b849f0e911199ba4fbf11 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli da41d2bb3406da22087b849f0e911199ba4fbf11 
(at node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine    : introspection-core da41d2bb3406da22087b849f0e911199ba4fbf11 (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt da41d2bb3406da22087b849f0e911199ba4fbf11 (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash    : da41d2bb3406da22087b849f0e911199ba4fbf11
Studio                  : 0.465.0
```

**Comment by chuahziyang** [comment]
I have done more investigating and it appears the NUXT3 is clearing the global object upon reload, is there any way to resolve this behaviour?

**Comment by austincrim** [comment]
Hey @chuahziyang, sorry for the delay in getting back to you.

If you are still experiencing this problem, I would suggest opening an issue in the Nuxt repo with the relevant details and research.

**Reply by vladi-stevanovic** [reply]
👋 Hello @chuahziyang did you perhaps have a chance to check Austin's recommendation? 

Let us know if you still have any questions, or if you've already solved it, we'd love to know how! It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
